client/src/main/java/io/pravega/client/netty/impl/ClientConnectionImpl.java

```diff
@@ -89,9 +90,17 @@ public void operationComplete(ChannelFuture future) {
         });
         // Work around for https://github.com/netty/netty/issues/3246
         eventLoop.execute(() -> {
-            channel.write(cmd, promise);
+            try {
+                if (!closed.get()) {
+                    channel.write(cmd, promise);
+                }
+            } catch (Exception e) {
+                throttle.release(dataLength);
+                channel.pipeline().fireExceptionCaught(e);
+                channel.close();
```

Comment:
We can directly invoke `io.pravega.client.netty.impl.ClientConnectionImpl#close`.